examples/move/walrus_dep/README.md

```diff
@@ -1,3 +1,11 @@
 # Example Move package depending on Walrus
 
 A simple example of depending on the Walrus Testnet Move package.
+
+You can build, test, and publish this with the following commands (from this directory):
+
+```sh
+sui move build
+sui move test
+sui client publish --skip-dependency-verification
```

Comment:
Probably not a good solution. Without this I got the following error:
```
Failed to publish the Move module(s), reason: [warning] Multiple source verification errors found:

- Local dependency did not match its on-chain version at 9f992cc2430a1f442ca7a5ca7638169f5d5c00e0ebc3977a65e9ac6e497fe5ef::Walrus::staking_pool
- Local dependency did not match its on-chain version at 9f992cc2430a1f442ca7a5ca7638169f5d5c00e0ebc3977a65e9ac6e497fe5ef::Walrus::active_set
- Local dependency did not match its on-chain version at 9f992cc2430a1f442ca7a5ca7638169f5d5c00e0ebc3977a65e9ac6e497fe5ef::Walrus::staking_inner
- Local dependency did not match its on-chain version at 9f992cc2430a1f442ca7a5ca7638169f5d5c00e0ebc3977a65e9ac6e497fe5ef::Walrus::staking
- Local version of dependency 0000000000000000000000000000000000000000000000000000000000000001::uq32_32 was not found.
- Local version of dependency 9f992cc2430a1f442ca7a5ca7638169f5d5c00e0ebc3977a65e9ac6e497fe5ef::wal_exchange was not found.

This may indicate that the on-chain version(s) of your package's dependencies may behave differently than the source version(s) your package was built against.

Fix this by rebuilding your packages with source versions matching on-chain versions of dependencies, or ignore this warning by re-running with the --skip-dependency-verification flag.
```

Reply:
I think we may need to add the published address to the move.toml. I can look into this.

Reply:
Which sui version do you have installed? 

I'm consistently getting a different error (tried with `testnet-v1.35.0` and with `testnet-v1.36.2`). 
With `1.35.0`, I'm getting
```
Failed to build Move modules: Processing dependency 'Walrus' of 'walrus_dep'

Caused by:
    Conflicting assignments for address 'walrus': '0x9f992cc2430a1f442ca7a5ca7638169f5d5c00e0ebc3977a65e9ac6e497fe5ef' and '0x0'.. 
```

and with `1.36.2`, I'm getting

```
Failed to publish the Move module(s), reason: Package dependency "WAL" does not specify a published address (the Move.toml manifest for "WAL" does not contain a 'published-at' field, nor is there a 'published-id' in the Move.lock).
Package dependency "Walrus" does not specify a published address (the Move.toml manifest for "Walrus" does not contain a 'published-at' field, nor is there a 'published-id' in the Move.lock).
If this is intentional, you may use the --with-unpublished-dependencies flag to continue publishing these dependencies as part of your package (they won't be linked against existing packages on-chain).
```

Reply:
Ah, I didn't realize that the dependency for `Walrus` was set to the main branch on git 🤦‍♂️. After changing this, I'm getting the same error as you with both versions.

I had two suspicions where the problem could originate, but neither of them seems to solve the problem.
Namely, I thought that the move compiler always uses the latest version, even for dependencies that specify a different one in `Move.lock`, which results in slightly different bytecode that doesn't match the on-chain version. However, using sui at version 1.35.0 doesn't fix the error.

Another issue could have been that the bytecode differs because `wal_exchange` as dependency of `walrus` is not included.  But adding the `wal_exchange` to the contracts also doesn't fix the error.

I think keeping the `--skip-dependency-verification` is probably best to make sure that we have a robust example.